### PR TITLE
BUILD/CONFIG: added content for basic pc files

### PR DESCRIPTION
**Pull request by hoopoepg**
- added libraries and dependencies for basic PC files: cma/knem/fuse
- updated PC files to use new macro for version definition

### CODE REVIEW

File: ucx.pc.in [hunk 1]
```diff
@@ -7,5 +7,7 @@ includedir = @includedir@
 Name: @PACKAGE@
 Description: Unified Communication X Library
 Version: @VERSION@
-Cflags: -I${includedir}
+Cflags: -I${includedir} 
+Cflags.private: -I${includedir}
```
Comment (brminich) [inline ucx.pc.in:11]: does `Cflags.private` really exists? What would that mean if headers are needed for build regardless of linking scenario?

File: src/ucs/vfs/fuse/ucx-fuse.pc.in [hunk 2]
```diff
@@ -1,5 +1,11 @@
...
+
 Name: @PACKAGE@-fuse
 Description: Unified Communication X Library FUSE module
-Version: @MAJOR_VERSION@.@MINOR_VERSION@
+Version: @VERSION@
 Libs:
-Libs.private:
+Libs.private: -L${libdir} -Wl,--whole-archive,-lucs_fuse,--no-whole-archive 
+Requires: fuse3
+Requires.private: fuse3
```
Comment (brminich) [inline src/ucs/vfs/fuse/ucx-fuse.pc.in:11]: is it needed even though it is identical to `requires`?

File: ucx.pc.in (hunk 1 above)
Comment (hoopoepg) [reply ucx.pc.in:11]: http://manpages.ubuntu.com/manpages/impish/man5/pc.5.html

File: src/ucs/vfs/fuse/ucx-fuse.pc.in (hunk 2 above)
Comment (hoopoepg) [reply src/ucs/vfs/fuse/ucx-fuse.pc.in:11]: as I can see yes. in any case pkg-config removes duplicates

File: ucx.pc.in [hunk 3]
```diff
@@ -9,3 +9,4 @@ Description: Unified Communication X Library
 Version: @VERSION@
 Cflags: -I${includedir}
 Libs: -L${libdir} -lucs -luct -lucp
+Libs.private: -lucm -Wl,--undefined=ucm_init -Wl,--undefined=ucs_init -Wl,--undefined=uct_init -Wl,--undefined=ucp_global_init @BFD_LIBS@ @BFD_LDFLAGS@ @BFD_DEPS@
```
Comment (yosefe) [inline ucx.pc.in:12]: can pkg-config take "ucs" dependency from UCS pkg-config file, and take "ucm" dependency from ucs pkg-config?
and have separate pkg-config from ucs, ucs, ucp, ucm
these are the deps:
- ucp->{uct, ucs, ucm}
- uct->{ucs, ucm}
- ucs->{ucm}

Comment (hoopoepg) [reply ucx.pc.in]: > can pkg-config take "ucs" dependency from UCS pkg-config file, and take "ucm" dependency from ucs pkg-

done

File: src/ucm/ucx-ucm.pc.in [hunk 4]
```diff
@@ -0,0 +1,12 @@
+prefix = @prefix@
```
Comment (yosefe) [inline src/ucm/ucx-ucm.pc.in:1]: Pls add NVIDIA copyright in all pkg-config files

Comment (hoopoepg) [reply src/ucm/ucx-ucm.pc.in:7]: > Pls add NVIDIA copyright in all pkg-config files

added

**Review by yosefe — APPROVED** [review]

**Review by brminich — APPROVED** [review]

**Comment by yosefe** [comment]
@hoopoepg pls squash

**Comment by hoopoepg** [comment]
@yosefe ok to merge?

**Review by yosefe — APPROVED** [review]

**Pull request merged**